merfi/backends/rpm_sign.py

```diff
@@ -39,6 +40,16 @@ def clear_sign(self, path, command):
         """
         logger.info('signing: %s' % path)
         out, err, code = util.run_output(command)
+        # Sometimes rpm-sign will fail with this error. I've opened a ticket
+        # with RH secalert. For now, sleep and retry as a workaround.
+        known_failure = "ERROR: unhandled exception occurred: ('')."
+        tries = 1
+        while known_failure in err and tries < 10:
+            logger.warning('hit known rpm-sign failure.')
+            tries += 1
+            logger.warning('sleeping, running try #%d in 30 seconds.' % tries)
+            sleep(30)
```

Comment:
why 30 seconds? sounds like a long time to just retry? is there any context as to why this was chosen over less/greater sleeps?

Reply:
Good question. I have very little data on how much sleep time is appropriate, because I only hit this failure in Jenkins about once a month. I just picked this out of a hat, trying to get more data.

From what I remember, the condition clears up when I hit "rebuild" in Jenkins, about 10-15 minutes later...

... although today is an exception, because I've hit the problem twice, 25 minutes apart. Maybe today's failure is a bigger server-side problem or something. But the STDERR text is the same every time that I have seen.

Reply:
Another data point, I hit "rebuild" a third time in Jenkins today, and 50 minutes after the last failure, it succeeded without errors (ie. without having to enter this loop at all)

Reply:
No problem. I was just thinking why would it need to wait to retry, but this is fine.